File: api/src/main/java/com/kjj/api/entity/store/StoreState.java
package com.kjj.api.entity.store;

import com.kjj.api.dto.store.StoreOffDto;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Index;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;

@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class StoreState {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    private Store store;

    @Index(name = "store_state_date_index")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate date;
    private Boolean off;
    private String name;

    public void setOff(StoreOffDto dto) {
        off = dto.isOff();
        name = dto.getName();
    }

    public static StoreState createNewOffStoreState(Store store, LocalDate date, StoreOffDto dto) {
        return new StoreState(
                null,
                store,
                date,
                dto.isOff(),
                dto.getName()
        );
    }
}

File: store/src/main/kotlin/com/kjj/store/entity/StoreState.kt
package com.kjj.store.entity

import com.kjj.store.dto.StoreOffDto
import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(indexes = arrayOf(
        Index(name = "store_state_date_index", columnList = "date"))
)
class StoreState(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?,
    @ManyToOne(fetch = FetchType.LAZY)
    val store: Store,

    val date: LocalDate,
    var off: Boolean,
    var name: String
) {
    companion object {
        fun createNewOffStoreState(store: Store, date: LocalDate, dto: StoreOffDto): StoreState {
            return StoreState(
                null,
                store,
                date,
                dto.off,
                dto.name
            )
        }
    }

    fun setOff(dto: StoreOffDto) {
        off = dto.off
        name = dto.name
    }
}